### PR TITLE
lowercase hex node ID

### LINKED ISSUE #971: `node::Id` changes values to capital letters in hex encoding
**Steps to reproduce**

The `node::Id` type (used in e.g. `TendermintConfig` via `net::Address`) rewrites values to capital letters when parsed with `FromStr` and serialized with `Display`. When used with e.g. the persistent_peers field in the p2p config, the capitalized node ID then do not match on connection authentication and hence cause failures.

**What's the definition of "done" for this issue?**

The `node::Id` shouldn't modify the values parsed from raw strings.

### PR DESCRIPTION
**Pull request by tzemanovic**
closes #971 

* [x] Referenced an issue explaining the need for the change
* [x] Updated all relevant documentation in docs
* [x] Updated all code comments where relevant
* [x] ~Wrote~ updated tests
* [x] Added entry in `.changelog/`

### CODE REVIEW

File: .changelog/unreleased/bug-fixes/1016-lowercase-node-id.md [hunk 1]
```diff
@@ -0,0 +1,2 @@
+- `[tendermint]` The `tendermint::node::Id` `Display` implementation now prints the hexadecimal string in lowercase ([#971](https://github.com/informalsystems/tendermint-
+  rs/issues/971))
```
Comment (thanethomson) [inline .changelog/unreleased/bug-fixes/1016-lowercase-node-id.md:2]:
```suggestion
- `[tendermint]` The `tendermint::node::Id` `Display` implementation now prints the hexadecimal string in lowercase ([#971](https://github.com/informalsystems/tendermint-rs/issues/971))
```

Was this automatically generated by `unclog`? If so I'll have to fix it there. Seems like it was wrapped at the `-` in `tendermint-rs`.

Comment (romac) [reply .changelog/unreleased/bug-fixes/1016-lowercase-node-id.md:2]: Yes, in my experience `unclog` sometimes break lines even in the middle of a URL.

Comment (thanethomson) [reply .changelog/unreleased/bug-fixes/1016-lowercase-node-id.md:2]: Yeah I'm using the `textwrap` crate and I thought I'd configured it to handle URLs properly. Will look into it.

Comment (tzemanovic) [reply .changelog/unreleased/bug-fixes/1016-lowercase-node-id.md]: I added this manually, thanks for this fixup!

**Review by thanethomson — APPROVED** [review]

**Pull request merged**